protected/humhub/modules/stream/models/filters/ModuleStreamFilter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\stream\models\filters;
+
+use humhub\models\Setting;
+use humhub\modules\content\components\ContentContainerActiveRecord;
+use humhub\modules\content\components\ContentContainerModule;
+use humhub\modules\content\models\ContentContainerModuleState;
+use humhub\modules\space\models\Space;
+use humhub\modules\user\models\User;
+use ReflectionClass;
+use Yii;
+
+class ModuleStreamFilter extends StreamQueryFilter
+{
+    public function apply(): void
+    {
+        if ($this->streamQuery->container instanceof ContentContainerActiveRecord) {
```

Comment:
@funkycram My original solution was this:

```php
$this->excludeContainersWithNotAvailableModule(Space::class);
if (!$this->streamQuery->container instanceof Space) {
    $this->excludeContainersWithNotAvailableModule(User::class);
}
```

but you made this as:

```php
if ($this->streamQuery->container instanceof ContentContainerActiveRecord) {
    $containerClass = $this->streamQuery->container instanceof Space ? Space::class : User::class;
    $this->excludeContainersWithNotAvailableModule($containerClass);
}
```

I would like to explain why I suggessted the solution.
The stream can be called from 3 sides:

1. Dashboard, here `$this->streamQuery->container === null` because we display entries from all spaces and users,
2. Space wall, here the container is `Space::class`, and here we display entries only from the Space,
3. User profile wall, here the container is `User::class`, and here we display enrties of the User also some entries may go from Spaces where the User posted them.

Correct me if I am wrong.
If I am right then my original solutions should be used instead of current, because:

- the Space filter `$this->excludeContainersWithNotAvailableModule(Space::class)` must be applied for all 3 points.
- the User filter `$this->excludeContainersWithNotAvailableModule(User::class)` should be applied only for 1st and 3rd cases when the stream container is `null` or `User::class`, it is why I use there the condition `!$this->streamQuery->container instanceof Space`.

Could please test these 3 cases(Dashboard, Space, User wall stream) to be sure all excluded as expected?
Thank you!

Reply:
Done in commit https://github.com/humhub/humhub/pull/6499/commits/05cbb7257374231744c67b34804adc78e241bfae